internal/errgroup/group_test.go

```diff
@@ -421,87 +426,157 @@ func Test_group_Go(t *testing.T) {
 	type args struct {
 		f func() error
 	}
-	type fields struct {
-		egctx            context.Context
-		cancel           func()
-		wg               sync.WaitGroup
-		limitation       chan struct{}
-		enableLimitation atomic.Value
-		cancelOnce       sync.Once
-		mu               sync.RWMutex
-		emap             map[string]struct{}
-		errs             []error
-		err              error
-	}
-	type want struct {
+	type generator struct {
+		do func() *group
 	}
 	type test struct {
 		name       string
 		args       args
-		fields     fields
-		want       want
-		checkFunc  func(want) error
+		generator  generator
+		checkFunc  func() error
 		beforeFunc func(args)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want) error {
+	defaultCheckFunc := func() error {
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           f: nil,
-		       },
-		       fields: fields {
-		           egctx: nil,
-		           cancel: nil,
-		           wg: sync.WaitGroup{},
-		           limitation: nil,
-		           enableLimitation: nil,
-		           cancelOnce: sync.Once{},
-		           mu: sync.RWMutex{},
-		           emap: nil,
-		           errs: nil,
-		           err: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           f: nil,
-		           },
-		           fields: fields {
-		           egctx: nil,
-		           cancel: nil,
-		           wg: sync.WaitGroup{},
-		           limitation: nil,
-		           enableLimitation: nil,
-		           cancelOnce: sync.Once{},
-		           mu: sync.RWMutex{},
-		           emap: nil,
-		           errs: nil,
-		           err: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			var calledCnt int32
+
+			var enableLimitation atomic.Value
+			enableLimitation.Store(true)
+
+			egctx, cancel := context.WithCancel(context.Background())
+			cancel()
+
+			g := &group{
+				egctx:            egctx,
+				limitation:       make(chan struct{}),
+				enableLimitation: enableLimitation,
+			}
+
+			return test{
+				name: "f is not called when reached limit and cancel g.egctx",
```

Comment:
I think we also need to test the limitation function (for example the limit is 3, and 5 `Go(f func error)` is executed and only 3 can be executed at first, any after 1 is completed the 4th func will execute)